src/Process/PlaceOnGroundProcessing.js

```diff
@@ -0,0 +1,126 @@
+import * as THREE from 'three';
+import Coordinates from '../Core/Geographic/Coordinates';
+import Extent from '../Core/Geographic/Extent';
+
+function tileAt(pt, tile) {
+    if (tile.extent) {
+        if (!tile.extent.pointIsInside(pt)) {
+            return undefined;
+        }
+
+        for (const c of tile.children) {
+            const t = tileAt(pt, c);
+            if (t) {
+                return t;
+            }
+        }
+        if (tile.materials[0].textures[0][0].coords.zoom > -1) {
+            return tile;
+        }
+        return undefined;
+    }
+}
+
+let _canvas;
+function _readTextureValueAt(layer, texture, x, y) {
+    if (texture.image.data) {
+        return texture.image.data[y * texture.image.width + x];
+    } else {
+        if (!_canvas) {
+            _canvas = document.createElement('canvas');
+            _canvas.width = 1;
+            _canvas.height = 1;
+        }
+        const ctx = _canvas.getContext('2d');
+        ctx.drawImage(texture.image, x, y, 1, 1, 0, 0, 1, 1);
+        const d = ctx.getImageData(0, 0, 1, 1);
+        return THREE.Math.lerp(
+            layer.materialOptions.colorTextureElevationMinZ,
+            layer.materialOptions.colorTextureElevationMaxZ,
+            d.data[0] / 255);
+    }
+}
+
+function _updateSinglePoint(referenceCrs, layer, nodes, crs, toModify) {
+    const pt = new Coordinates(referenceCrs, toModify).as(crs);
+    pt._values[2] = 0;
+
+    let smallest = null;
+    for (const n of nodes) {
+        smallest = tileAt(pt, n);
+        if (smallest) {
+            break;
+        }
+    }
+
+    if (smallest) {
+        const src = smallest.materials[0].textures[0][0];
+        const width = src.image.width;
+        const height = src.image.height;
+
+        const tmp = new Extent(pt.crs, pt, pt);
+
+        const goUp = Math.log2(1.0 / smallest.materials[0].offsetScale[0][0].z);
+        for (let i = 0; i < goUp; i++) {
```

Comment:
Factorize in `/src/Process/TiledNodeProcessing.js` : 
`getElevationFromTile(tile, coords)`